doc/connectivity/networking/dsa.rst

```diff
@@ -0,0 +1,125 @@
+.. _dsa:
+
+Distributed Switch Architecture (DSA)
+#####################################
+
+.. contents::
+    :local:
+    :depth: 2
+
+Distributed Switch Architecture (DSA) is not something new. It has been a mature
+subsystem in Linux for many years. This document just skips the background,
+terms and any knowledge related description, as user may find all of these in
+`Linux DSA documentation`_.
+
+
+Architecture
+************
+
+Same with Linux, from network device perspective the DSA architecture is as below.
+
+.. code-block:: console
+
+                Unaware application
+              opens and binds socket
+                       |  ^
+                       |  |
+           +-----------v--|--------------------+
+           |+------+ +------+ +------+ +------+|
+           || swp0 | | swp1 | | swp2 | | swp3 ||
+           |+------+-+------+-+------+-+------+|
+           |          DSA switch driver        |
+           +-----------------------------------+
+                         |        ^
+            Tag added by |        | Tag consumed by
+           switch driver |        | switch driver
+                         v        |
+           +-----------------------------------+
+           | Unmodified host interface driver  | Software
+   --------+-----------------------------------+------------
+           |       Host interface (eth0)       | Hardware
+           +-----------------------------------+
+                         |        ^
+         Tag consumed by |        | Tag added by
+         switch hardware |        | switch hardware
+                         v        |
+           +-----------------------------------+
+           |               Switch              |
+           |+------+ +------+ +------+ +------+|
+           || swp0 | | swp1 | | swp2 | | swp3 ||
+           ++------+-+------+-+------+-+------++
```

Comment:
I think we could the rx/tx diagram in https://github.com/zephyrproject-rtos/zephyr/discussions/87091, but not use this diagram which is from Linux kernel document.

Reply:
Okay, will replace it.
Thanks.